Add strformat support for Complex numbers

AngelEzquerra opened the pull request:
Before this change strformat used the generic formatValue function for Complex numbers. This meant that it was not possible to control the format of the real and imaginary components of the complex numbers.

With this change this now works:
```nim
import std/[complex, strformat]
let c = complex(1.05000001, -2.000003)
echo &"{c:g}"
# You now get: (1.05, -2)
# while before you'd get a ValueError exception (invalid type in format string for string, expected 's', but got g)
```

The only small drawback of this change is that I had to import complex from strformat. I hope that is not a problem.

AngelEzquerraAtKeysight committed:
Add strformat support for Complex numbers

Before this change strformat used the generic formatValue function for Complex numbers. This meant that it was not possible to control the format of the real and imaginary components of the complex numbers.

With this change this now works:
```nim
import std/complex
let c = complex(1.05000001, -2.000003)
echo &"{c:g}"
# You now get: (1.05, -2)
# while befor you'd get a ValueError exception (invalid type in format string for string, expected 's', but got g)
```

beef331 commented:
I personally do not see the benefit in adding non primitive formatters to `strformat`. `formatValue` can be defined anywhere as long as it's accessible inside the module that calls `&` or `fmt`. One could add it to `std/complex` instead which might make more sense, or a package that implements a formatter for a variety of mathematic types (vectors, quaternions, matrices, ...)

AngelEzquerra commented:
> I personally do not see the benefit in adding non primitive formatters to `strformat`. `formatValue` can be defined anywhere as long as it's accessible inside the module that calls `&` or `fmt`. One could add it to `std/complex` instead which might make more sense, or a package that implements a formatter for a variety of mathematic types (vectors, quaternions, matrices, ...)

This was discussed in #internal and the same suggestion was made. I'll move the formatter to complex.nim.

AngelEzquerraAtKeysight committed:
Moved the complex formatValue procedure to complex.nim.

AngelEzquerraAtKeysight committed:
Add 'j' format specifier for complex numbers

When j is used, complex numbers are specified as complex "numbers" using the typical mathematical (A+Bj) format.
By default, the real and imaginary parts are formatted using the general ('g') format.
